cluster-autoscaler/core/scale_up.go

```diff
@@ -96,8 +94,10 @@ func ScaleUp(context *context.AutoscalingContext, clusterStateRegistry *clusters
 	podsRemainUnschedulable := make(map[*apiv1.Pod]bool)
 	expansionOptions := make([]expander.Option, 0)
 
-	if context.AutoscalingOptions.NodeAutoprovisioningEnabled {
-		nodeGroups, nodeInfos = addAutoprovisionedCandidates(context, nodeGroups, nodeInfos, unschedulablePods)
+	var errProc error
```

Comment:
Is there any reason to declare this variable separately?

Reply:
I don't want to use := operator, because the code relies on overriding existing variables. If we decided that we need to move this call into an if statement for whatever reason, we would suddenly create temporary nodeGroups and nodeInfos variables. I've debugged exactly this kind of error too many times already (and it's the reason why I, personally, consider := operator the worst design choice of golang).

Still, I agree this variable declaration is kinda ugly. If you have a more pretty way of achieving the same, I'm happy to change it.

Reply:
Just wanted to point out that adding handling for nil values required moving inside if statement, as described in my comment above :) Operator := is pure evil.